app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,36 @@
+class OrdersController < ApplicationController
+  def show
+    if !current_user || current_user != (@order = Order.find(params[:id])).user
+      @access_denied = true
+    end
+  end
+
+  def new
+    @event = Event.find(params[:event_id])
+    if !@event.has_expired? && !current_user
+      store_location_and_require_login
+    end
+    @order = Order.build_from_event(@event)
+  end
+
+  def create
+    @order = Order.new(order_params)
+    if @order.save
```

Comment:
I think we had this style in another place `if (@order = Order.new(order_params)).save` so let's use that.